src/ParallelAlgorithms/ApparentHorizonFinder/ComputeVarsToInterpolateToTarget.cpp

```diff
@@ -0,0 +1,259 @@
+  volume_vars_storage->vars_to_interpolate_to_target.initialize(
+      volume_vars_storage->source_vars.number_of_grid_points());
+
+  const auto& mesh = volume_vars_storage->mesh;
+  const auto& block = domain.blocks()[element_id.block_id()];
+  if (block.is_time_dependent()) {
+    if constexpr (std::is_same_v<Fr, ::Frame::Grid>) {
+      ElementMap<3, Fr> map_logical_to_grid{
+          element_id, block.moving_mesh_logical_to_grid_map().get_clone()};
+      const auto logical_coords = logical_coordinates(mesh);
+      const auto jac_grid_to_inertial =
+          block.moving_mesh_grid_to_inertial_map().jacobian(
+              map_logical_to_grid(logical_coords), time.id, functions_of_time);
+      const auto invjac_logical_to_grid =
+          map_logical_to_grid.inv_jacobian(logical_coords);
+
+      compute_vars_to_interpolate_to_target_impl(
+          volume_vars_storage, jac_grid_to_inertial, invjac_logical_to_grid);
+    } else if constexpr (std::is_same_v<Fr, ::Frame::Distorted>) {
+      ASSERT(block.has_distorted_frame(),
+             "Block doesn't have a distorted frame but this horizon is in the "
+             "distorted frame.");
+      const domain::CoordinateMaps::Composition
+          element_logical_to_distorted_map{
+              domain::element_to_block_logical_map(element_id),
+              block.moving_mesh_logical_to_grid_map().get_clone(),
+              block.moving_mesh_grid_to_distorted_map().get_clone()};
+      const domain::CoordinateMaps::Composition element_logical_to_grid_map{
+          domain::element_to_block_logical_map(element_id),
+          block.moving_mesh_logical_to_grid_map().get_clone()};
+      const auto logical_coords = logical_coordinates(mesh);
+      const auto jac_distorted_to_inertial =
+          block.moving_mesh_distorted_to_inertial_map().jacobian(
+              element_logical_to_distorted_map(logical_coords, time.id,
+                                               functions_of_time),
+              time.id, functions_of_time);
+      const auto invjac_logical_to_distorted =
+          element_logical_to_distorted_map.inv_jacobian(logical_coords, time.id,
+                                                        functions_of_time);
+
+      compute_vars_to_interpolate_to_target_impl(volume_vars_storage,
+                                                 jac_distorted_to_inertial,
+                                                 invjac_logical_to_distorted);
+    } else {
+      compute_vars_to_interpolate_to_target_impl(volume_vars_storage);
+    }
+  } else {
+    // No frame transformations needed, since the maps are time-independent
+    // and therefore all the frames are the same.
+    //
+    // The frame tags may be different, but the source vars should all be in the
+    // Inertial frame, so we force the inertial frame (True template).
+    compute_vars_to_interpolate_to_target_impl<Fr, true>(volume_vars_storage);
```

Comment:
[optional] Reorder the template parameters on the impl so the frame can be inferred here.